WordPress/Classes/ViewRelated/Me/My Profile/Change Username/ChangeUsernameViewController.swift

```diff
@@ -15,7 +15,8 @@ class ChangeUsernameViewController: SignupUsernameTableViewController {
         }
         return saveItem
     }()
-    private var changeUsernameAction: UIAlertAction?
+
+    private weak var confirmationController: UIAlertController?
```

Comment:
It's unfortunate that `UIAlertController` is such a weird bit of API – it'd be great to push down the `TextViewDidChange` stuff into a `UIAlertController` subclass (if such a thing were allowed, which it's not for some reason).

I really wonder if adjusting the UI (to present a full-screen page instead of a jammed-up confirmation step) would make this a lot simpler code-wise? Seems like a win on both fronts, tbh.

Reply:
I created https://github.com/wordpress-mobile/WordPress-iOS/pull/21006 as an alternative to this PR, using a static function instead of a subclass. But I think the idea is similar: encapsulating the logic within `UIAlertController`.